server/portal/apps/projects/management/commands/projects_id.py

```diff
@@ -99,29 +75,51 @@ def add_arguments(self, parser):
             '--update',
             action='store',
             type=int,
-            help='Integer to update saved DB value.'
+            help='Update project id DB value.'
+        )
+        parser.add_argument(
+            '--update-using-storage-system-id',
+            action='store_true',
+            help='Update project id DB value using value derived from latest storage system project id.'
+        )
+        parser.add_argument(
+            '--ignore-large-project-ids',
+            action='store',
+            type=int,
+            help='Ignore project ids larger than a certain value'
         )
 
     def handle(self, *args, **options):
         """Handle command."""
+        ignore_large_ids = options["ignore_large_project_ids"]
+        if ignore_large_ids:
+            self.stdout.write('NOTE(!!!!): Ignoring project ids >= {} when '
+                              'processing/updating the storage systems and directories'.format(ignore_large_ids))
+
+        latest_storage_system_id = Command.get_latest_project_storage(ignore_above_value=ignore_large_ids)
+        latest_project_id = Command.get_latest_project_directory(ignore_above_value=ignore_large_ids)
+
+        if latest_storage_system_id == -1:
+            self.stdout.write('There are no project storage systems.')
+        if latest_project_id == -1:
+            self.stdout.write('There are no project directories.')
+
+        self.stdout.write('Latest storage system project id: {}'.format(latest_storage_system_id))
+        self.stdout.write('Latest directory project id: {}'.format(latest_project_id))
+
+        try:
+            with transaction.atomic():
+                model_project_id = ProjectId.objects.select_for_update().latest('last_updated').value
+            self.stdout.write('Latest project id in ProjectId model: {}'.format(model_project_id))
+        except ObjectDoesNotExist:
+            self.stdout.write('Latest project id in ProjectId mode: None')
+            if (options.get('update') or options.get('update_using_storage_system_id')):
+                ProjectId.objects.create(value=1).save()
```

Comment:
Wouldn't we want to update to a different value here depending on `update` or `update_using_storage_system_id`?

Reply:
will refactor this. i'm working around the ProjectId.update() needing a value before it can update but this was the wrong way to approach it; will fix it.

Reply:
Refactored in 5d1e90c